### PR TITLE
idle cleanup some code

### PR DESCRIPTION
**Pull request by CrockAgile**
idle cleanup some code

### CODE REVIEW

File: src/slice_iter.rs [hunk 1]
```diff
@@ -1,44 +0,0 @@
-/// Generic Iterator wrapper of slice
```
Comment (CrockAgile) [inline src/slice_iter.rs:1]: I'll be honest. I have no idea what these slice iterators were for 🤔 maybe wrote them before the API was changed to returning `impl Iterator` but now they can be removed ? a mystery to me now

Comment (shnewto) [reply src/slice_iter.rs:1]: haha luckily you made some good comments in the PRs that informed the changes #94 and #102

Comment (CrockAgile) [reply src/slice_iter.rs:1]: right but the slice iterators also implement size_hint so I am still at a loss why I did it this way

Comment (shnewto) [reply src/slice_iter.rs:1]: lol 🤷

**Review by shnewto — APPROVED** [review]
looks good!

**Pull request merged**